base/samples/BodyState.hpp

```diff
@@ -0,0 +1,337 @@
+        }
+
+        inline base::Position& position()
+        {
+            return this->pose.translation;
+        }
+
+        /** A read-only expression of the rotation **/
+        inline const base::Quaterniond& orientation() const
+        {
+            return this->pose.orientation;
+        }
+
+        inline base::Quaterniond& orientation()
+        {
+            return this->pose.orientation;
+        }
+
+        inline const base::Vector3d& linear_velocity() const
+        {
+            return this->velocity.vel;
+        }
+
+        inline base::Position& linear_velocity()
+        {
+            return this->velocity.vel;
+        }
+
+        inline const base::Vector3d& angular_velocity() const
+        {
+            return this->velocity.rot;
+        }
+
+        inline base::Position& angular_velocity()
+        {
+            return this->velocity.rot;
+        }
+
+
+        /** A read-only expression of the pose covariance **/
+        inline const base::Matrix6d& cov_pose() const
+        {
+            return this->pose.cov;
+        }
+
+        inline base::Matrix6d& cov_pose()
+        {
+            return this->pose.cov;
+        }
+
+        /** A read-only expression of the rotation covariance **/
+        inline const base::Matrix3d cov_orientation() const
+        {
+            return this->pose.getOrientationCov();
+        }
+
+        inline void cov_orientation(const base::Matrix3d& cov)
+        {
+            return this->pose.setOrientationCov(cov);
+        }
+
+        /** A read-only expression of the position covariance **/
+        inline const base::Matrix3d cov_position() const
+        {
+            return this->pose.getTranslationCov();
+        }
+
+        inline void cov_position(const base::Matrix3d& cov)
+        {
+            return this->pose.setTranslationCov(cov);
+        }
+
+        /** A read-only expression of the velocity covariance **/
+        inline const base::Matrix6d& cov_velocity() const
+        {
+            return this->velocity.cov;
+        }
+
+        inline base::Matrix6d& cov_velocity()
+        {
+            return this->velocity.cov;
+        }
+
+        /** A read-only expression of the linear velocity covariance **/
+        inline const base::Matrix3d cov_linear_velocity() const
+        {
+            return this->velocity.getLinearVelocityCov();
+        }
+
+        inline void cov_linear_velocity(const base::Matrix3d& cov)
+        {
+            return this->velocity.setLinearVelocityCov(cov);
+        }
+
+        /** A read-only expression of the angular velocity covariance **/
+        inline const base::Matrix3d cov_angular_velocity() const
+        {
+            return this->velocity.getAngularVelocityCov();
+        }
+
+        inline void cov_angular_velocity(const base::Matrix3d& cov)
+        {
+            return this->velocity.setAngularVelocityCov(cov);
+        }
+
+        static BodyState unknown()
```

Comment:
I would suggest naming them Invalid() and Unknown() (i.e. using uppercase) to match Eigen's Zero, Identity, ...